core/vm/contracts.go

```diff
@@ -1258,3 +1288,150 @@ func (c *blsSignatureVerify) Run(input []byte) ([]byte, error) {
 
 	return big1.Bytes(), nil
 }
+
+// verifyHeader implements bsc header verification precompile.
+type verifyHeader struct{}
+
+// RequiredGas returns the gas required to execute the pre-compiled contract.
+func (c *verifyHeader) RequiredGas(input []byte) uint64 {
+	msgAndSigLength := msgHashLength + signatureLength
+	inputLen := uint64(len(input))
+	if inputLen <= msgAndSigLength ||
+		(inputLen-msgAndSigLength)%singleBlsPubkeyLength != 0 {
+		return params.BlsSignatureVerifyBaseGas
+	}
+	pubKeyNumber := (inputLen - msgAndSigLength) / singleBlsPubkeyLength
+	return params.BlsSignatureVerifyBaseGas + pubKeyNumber*params.BlsSignatureVerifyPerKeyGas
+}
+
+var (
+	bscHeaderType, _ = abi.NewType("tuple", "", []abi.ArgumentMarshaling{
+		{Name: "ParentHash", Type: "bytes32"},
+		{Name: "UncleHash", Type: "bytes32"},
+		{Name: "Coinbase", Type: "address"},
+		{Name: "Root", Type: "bytes32"},
+		{Name: "TxHash", Type: "bytes32"},
+		{Name: "ReceiptHash", Type: "bytes32"},
+		{Name: "Bloom", Type: "bytes"},
+		{Name: "Difficulty", Type: "uint64"},
+		{Name: "Number", Type: "uint64"},
+		{Name: "GasLimit", Type: "uint64"},
+		{Name: "GasUsed", Type: "uint64"},
+		{Name: "Time", Type: "uint64"},
+		{Name: "Extra", Type: "bytes"},
+		{Name: "MixDigest", Type: "bytes32"},
+		{Name: "Nonce", Type: "bytes8"},
+	})
+
+	bscHeaderArgs = abi.Arguments{
+		{Type: bscHeaderType},
+	}
+
+	extraSeal = 65
+)
+
+type BscHeaderStruct struct {
+	ParentHash  [32]byte
+	UncleHash   [32]byte
+	Coinbase    common.Address
+	Root        [32]byte
+	TxHash      [32]byte
+	ReceiptHash [32]byte
+	Bloom       []byte
+	Difficulty  uint64
+	Number      uint64
+	GasLimit    uint64
```

Comment:
I doubt this is the latest block header structure, becase EIP1559 is enabled, atleast the `baseFee` is part of the header.

Reply:
Done